File: Paleon/src/com/paleon/astar/Node.java
package com.paleon.astar;

import com.paleon.utils.IHeapItem;
import com.paleon.utils.MathUtils;

public class Node<T> extends IHeapItem<Node<T>> {

	public T data;
	
	public int gCost;
	public int hCost;
	
	public Node<T> parent;
	
	private int heapIndex;
	
	public int getfCost() {
		return gCost + hCost;
	}

	@Override
	public int compareTo(Node<T> nodeToCompare) {
		int compare = MathUtils.compareTo(getfCost(), nodeToCompare.getfCost());
		if(compare == 0) {
			compare = MathUtils.compareTo(hCost, nodeToCompare.hCost);
		}
		return -compare;
		
	}

	@Override
	public int getHeapIndex() {
		return heapIndex;
	}

	@Override
	public void setHeapIndex(int heapIndex) {
		this.heapIndex = heapIndex;
	}
	
}

File: Technolithic/Technolithic/Pathfinding/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Technolithic
{
    public class Node<T> : IHeapItem<Node<T>>
    {
        public T Data;
        public int GCost;
        public int HCost;
        public Node<T> Parent;

        public Node()
        {

        }

        public int FCost
        {
            get
            {
                return GCost + HCost;
            }
        }

        public int HeapIndex
        {
            get; set;
        }

        public int CompareTo(Node<T> nodeToCompare)
        {
            int compare = FCost.CompareTo(nodeToCompare.FCost);
            if (compare == 0)
                compare = HCost.CompareTo(nodeToCompare.HCost);

            return -compare;
        }
    }
}
